xbmc/addons/interfaces/platforms/General.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ *      Copyright (C) 2005-2018 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "addons/interfaces/General.h"
+#include "addons/kodi-addon-dev-kit/include/kodi/General.h"
+
+using namespace kodi; // addon-dev-kit namespace
+
+namespace ADDON
+{
+
+char* Interface_General::get_platform_property(void* kodiBase, const char *name)
```

Comment:
I doubt that this stub will ever get substituted by other platforms. So what we've got here is an extension of the generic interface only used by a single platform.
Why not creating a platform specific interface for platform specific information.
Instead of GetPlatformProperty() it should be GetPlatformInterface()->GetWhatever()

Reply:
Reason is that I want to avoid ifdeffery in addon too. Using platform depenndent interface would require to use exactly this: ifdefs here and there.

Reply:
I don't see why a platform specific interface would require an ifdef.
Kodi has so many addons and supports a lot of platforms. If we allow every addon to create its specific interface methods, we'll end up in a mess.